dynamo-event-to-datacite-request/src/main/java/no/unit/nva/events/models/doi/PublicationMapper.java

```diff
@@ -0,0 +1,107 @@
+    private static final JsonPointer PUBLICATION_IDENTIFIER_POINTER
+        = JsonPointer.compile(ROOT + "/newImage/identifier/s");
+    private static final JsonPointer PUBLICATION_TYPE_POINTER = JsonPointer.compile(
+        ROOT + "/newImage/entityDescription/m/reference/m/publicationInstance/m/type/s");
+    private static final JsonPointer PUBLICATION_ENTITY_DESCRIPTION_POINTER = JsonPointer.compile(
+        ROOT + "/newImage/entityDescription/m");
+    private static final JsonPointer DOI_POINTER = JsonPointer.compile(
+        ROOT + "/newImage/entityDescription/m/reference/m/doi/s");
+    private static final JsonPointer MAIN_TITLE_POINTER = JsonPointer.compile(
+        ROOT + "/newImage/entityDescription/m/mainTitle/s");
+    private static final JsonPointer TYPE_POINTER = JsonPointer.compile(ROOT + "/newImage/type/s");
+    private static final JsonPointer INSTITUTION_OWNER_POINTER = JsonPointer.compile(ROOT + "/newImage/publisherId/s");
+
+    private PublicationMapper() {
+
+    }
+
+    /**
+     * Map to doi.Publication from a dynamo db stream record from nva_publication / nva_resources
+     *
+     * @param publicationIdPrefix                        prefix for a publication, from running environment
+     *                                                   (https://nva.unit.no/publication)
+     * @param dynamodbStreamRecordSerializedAsJsonString detail.dynamodb serialized as a string
+     * @return Publication doi.Publication
+     * @throws IOException on IO exception
+     */
+    public static Publication fromDynamodbStreamRecord(String publicationIdPrefix,
+                                                       String dynamodbStreamRecordSerializedAsJsonString)
+        throws IOException {
+        var record = JsonUtils.objectMapper.readTree(dynamodbStreamRecordSerializedAsJsonString);
+
+        var typeAttribute = textFromNode(record, TYPE_POINTER);
+        if (typeAttribute == null || !typeAttribute.equals(PUBLICATION_TYPE)) {
+            throw new IllegalArgumentException("Must be a dynamodb stream record of type Publication");
+        }
+        var publicationBuilder = PublicationBuilder.newBuilder();
+        publicationBuilder
+            .withId(URI.create(publicationIdPrefix + textFromNode(record, PUBLICATION_IDENTIFIER_POINTER)))
+            .withType(PublicationType.findByName(textFromNode(record, PUBLICATION_TYPE_POINTER)))
+            .withPublicationDate(new PublicationDate(record.at(PUBLICATION_ENTITY_DESCRIPTION_POINTER)))
+            .withTitle(textFromNode(record, MAIN_TITLE_POINTER))
+            .withInstitutionOwner(URI.create(textFromNode(record, INSTITUTION_OWNER_POINTER)))
+            .withContributor(toStream(record.at(CONTRIBUTORS_LIST_POINTER))
+                .map(PublicationMapper::extractContributor)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList()));
```

Comment:
refactor the logic inside the parenthesis to a method with a name such as `extractContributors`